addons/calendar/data/mail_template_data.xml

```diff
@@ -449,8 +449,9 @@
                         </t>
                         <t t-if="object.location">
                             <li>Location: <t t-out="object.location or ''">Bruxelles</t>
-                                (<a target="_blank"
-                                    t-attf-href="http://maps.google.com/maps?oi=map&amp;q={{object.location}}">View Map</a>)
+                                <a target="_blank"
+                                    t-if="object.location != object.appointment_type_id.location_id.name"
```

Comment:
And we should also do the trick with is_online here or it will also raise an error

Reply:
Okay, so when we have only calendar event (without appointment type), by using condition `not is_online`, it would always display the link even if `res.partner` does not have a `contact_address` because calendar.event does not have any field like we have `location_id` in 'appointment_type.py' on the basis of which location is computed. So just making sure what we would actually want here. 

_Reference images (Without address and without appointment type):_ 
[Template for: Date Updated](https://imgur.com/a/G8v04IH)
[Template for: Event Updated - random name](https://imgur.com/a/Ct7Mm6a)
[Template for: Event Updated - contact name](https://imgur.com/a/BC5OvvD)

Let me know what you think and if we want to handle this. I'll proceed accordingly...

Reply:
Indeed if it's not linked to an appointment type, we only have the field `location` of the calendar event.
This is fine, as it's the previous behavior and it's a value the user has filled in manually on the form view.